code/modules/resleeving/resleeving_sickness.dm

```diff
@@ -21,4 +21,28 @@
 	stacks = MODIFIER_STACK_EXTEND
 
 	on_created_text = "<span class='warning'>You feel slightly weak and unfocused.</span>"
-	on_expired_text = "<span class='notice'>You feel your strength and focus return to you.</span>"
\ No newline at end of file
+	on_expired_text = "<span class='notice'>You feel your strength and focus return to you.</span>"
+
+/datum/modifier/gory_devourment
+	name = "gory devourment"
+	desc = "You are being devoured! Dying right now would definitely be same as dying as food."
+	stacks = MODIFIER_STACK_EXTEND
+	hidden = TRUE
+	var/datum/mind/cached_mind = null
+
+/datum/modifier/gory_devourment/can_apply(var/mob/living/L)
+	if(L.stat == DEAD)
+		return FALSE
+	else
+		return TRUE
+
+/datum/modifier/gory_devourment/on_applied()
+	if(holder.ckey)
+		if(holder.mind)
+			cached_mind = holder.mind
+	return
+
+/datum/modifier/gory_devourment/on_expire()
+	if(holder.stat == DEAD)
+		cached_mind.vore_death = TRUE
```

Comment:
Shouldn't it be vore_death `FALSE` when it expires, if it expired before they died...?

Reply:
Its false by default, and that var is only meant to be set during or shortly after death anyway, so I do not think that is a problem.

Reply:
Oh nevermind, I see what you mean.